### PR TITLE
[App Service] `az appservice ase create-inbound-services`: Add support for Azure private DNS zone creation in ASEv3

### PR DESCRIPTION
**Pull request by madsd**
**Description**
ASEv3 support has been pending azure-mgmt-web update to 5.0.0+. With update to 6.1.0 we can unblock this scenario.

**Testing Guide**
az appservice ase create -n MyASEv3Name -g ASEv3ResourceGroup --vnet-name MyASEv3VirtualNetwork --subnet MyASEv3Subnet --kind asev3 --virtual-ip-type Internal
az appservice ase create-inbound-services -n MyASEv3Name -g ASEv3ResourceGroup --vnet-name MyASEv3VirtualNetwork --subnet MyASEv3Subnet

Result: Azure private DNS zone is created with the necessary inbound ILB ip DNS records configured and zone linked to vnet.

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

**Comment by panchagnula** [comment]
@StrawnSC could you review this please? Thanks!

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 1]
```diff
@@ -950,11 +950,12 @@ def load_arguments(self, _):
...
         c.argument('name', options_list=['--name', '-n'], help='Name of the app service environment',
                    local_context_attribute=LocalContextAttribute(name='ase_name', actions=[LocalContextAction.GET]))
-        c.argument('subnet', help='Name or ID of existing subnet for inbound traffic to ASEv3. \
+        c.argument('subnet', help='Name or ID of existing subnet for DNS Zone link. \
                    To create vnet and/or subnet use `az network vnet [subnet] create`')
         c.argument('vnet_name', help='Name of the vNet. Mandatory if only subnet name is specified.')
         c.argument('skip_dns', arg_type=get_three_state_flag(),
-                   help='Do not create Private DNS Zone and DNS records.')
+                   help='Do not create Private DNS Zone and DNS records.',
+                   deprecate_info=c.deprecate(expiration='3.0.0'))
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:958]: this (3.0.0) might be a while  - is this by design?

File: src/azure-cli/azure/cli/command_modules/appservice/appservice_environment.py [hunk 2]
```diff
@@ -152,11 +152,12 @@ def create_ase_inbound_services(cmd, resource_group_name, name, subnet, vnet_nam
         raise ValidationError('Private DNS Zone is not relevant for External ASE.')
 
     if ase.kind.lower() == 'asev3':
-        # pending SDK update (ase_client.get_ase_v3_networking_configuration(resource_group_name, name))
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/appservice_environment.py:155]: nit: remove?

**Comment by panchagnula** [comment]
@madsd do we need any mock tests here?

**Comment by StrawnSC** [comment]
Sorry for the delay in leaving a review. I'm in the process of testing out the command

**Review by StrawnSC — APPROVED** [review]
The functionality looks good to me -- the command ran successfully and created a private DNS zone when I tried it

**Review by wangzelin007 — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@madsd Could you please reply those comments from @panchagnula

**Comment by zhoxing-ms** [comment]
@madsd @panchagnula Do you think this PR can be merged at present? Please note that if it is not confirmed before tomorrow, this PR will not catch up with the release of this sprint and can only be postponed to the next sprint (about early May)

**Comment by zhoxing-ms** [comment]
@panchagnula  Confirmed in an email thread, so I merge this PR

**Pull request merged**